### ISSUE TITLE
foreach with static variables fails

### ISSUE THREAD

**Issue by grimmdude** [facebook/hhvm#5045]
It may not be the best practice, but this fails in HHVM:

```php
class Test
{
    private static $list = ['a','b','c','d'];
    private static $list_item;

    public static function iterate () 
    {
        foreach(self::$list as self::$list_item) {
            
        }
    }


}

```

**Comment by r3wt** [comment]
http://3v4l.org/VK7Xo

**Comment by nikic** [comment]
Probably an `emitClsIfSPropBase(e)` missing after https://github.com/facebook/hhvm/blob/679b5cca3e0ba72edf594916a63dc859635eb3d8/hphp/compiler/analysis/emitter.cpp#L8199.

**Comment by SiebelsTim** [comment]
Fixed in #5047
